src/mutations/events.ts

```diff
@@ -1,48 +1,78 @@
+import env from "dotenv";
 import { Event, User } from "../generated/prisma-client";
 import { resolveUserUsingJWT } from "../utils/resolveUser";
+import { uploadToS3, deleteFromS3 } from "../utils/S3";
+import { S3UploadResponse, S3DeleteResponse } from "../utils/S3/types";
 import assert from "assert";
 
-const createEvent = async (parent, data, ctx): Promise<Event> => {
+env.config();
+const bucket_suffix =
+  process.env.NODE_ENV === "development" ? "dev" : "production";
+const bucket_name = `rctechclub-raven/${bucket_suffix}`;
+
+const createEvent = async (
+  parent,
+  { title, start, end, venue, image_base64, description },
+  ctx
+): Promise<Event> | null => {
   const user: User | null = await resolveUserUsingJWT(ctx);
   assert.notStrictEqual(user, null, "No user login");
 
-  const title: string = data.title;
-  const start: Date = new Date(data.start);
-  const end: Date = new Date(data.end);
-  const venue: string = data.venue;
-  const image_base64: string = data.image_base64;
-  const description: string = data.description;
-
-  //image_url should be a link pointing to S3
-  //a utility function to be further implemented
-  const image_url = image_base64;
+  start = new Date(start);
+  end = new Date(end);
 
-  const event: Event = await ctx.prisma.createEvent({
+  let event: Event = await ctx.prisma.createEvent({
     title,
     start,
     end,
     venue,
-    image_url,
     description,
+    image_url: "",
     organiser: {
       connect: {
         id: user.id,
       },
     },
   });
 
+  //Store image_base64 to S3 bucket with filename <event.id>
+  //And retrieve image_url back to be stored in the database
+  const { id } = event;
+  const S3UploadResponse: S3UploadResponse = await uploadToS3({
+    image_base64,
+    file_name: id + ".jpg",
+    bucket_name,
+  });
+  const { isSuccessful, image_url } = S3UploadResponse;
+
+  //return null if unable to upload image to S3
+  if (!isSuccessful) return null;
```

Comment:
`null` is retuned if image upload fails, but `event` data seems to be stored into prisma database regardless, this means there's inconsistencies between the response received by client app, and the actual database

I think we need to decide what to do in this case.

Reply:
yeahh that's true, let's bring that to the meeting?

Reply:
what im thinking for now is we can delete the pre-created event if image upload fails and return `null` to the client. wdyt??

Reply:
seems like it

Reply:
alrightt, will make the changes